flow360/component/flow360_params/flow360_output.py

```diff
@@ -0,0 +1,323 @@
+    residual_sa: Optional[bool] = pd.Field(alias="residualSA")
+    coarsen_iterations: Optional[int] = pd.Field(alias="coarsenIterations")
+
+
+class Slice(Flow360BaseModel):
+    """:class:`NamedSlice` class"""
+
+    slice_normal: Coordinate = pd.Field(alias="sliceNormal")
+    slice_origin: Coordinate = pd.Field(alias="sliceOrigin")
+    output_fields: Optional[List[str]] = pd.Field(alias="outputFields")
+
+
+class Slices(Flow360SortableBaseModel):
+    """:class:`SelfNamedSlices` class"""
+
+    @classmethod
+    def get_subtypes(cls) -> list:
+        return [_GenericSliceWrapper.__fields__["v"].type_]
+
+    # pylint: disable=no-self-argument
+    @pd.root_validator(pre=True)
+    def validate_slice(cls, values):
+        """
+        root validator
+        """
+        return _self_named_property_validator(
+            values, _GenericSliceWrapper, msg="is not any of supported slice types."
+        )
+
+
+class _GenericSliceWrapper(Flow360BaseModel):
+    """:class:`_GenericMonitorWrapper` class"""
+
+    v: Slice
+
+
+class SliceOutput(OutputBase):
+    """:class:`SliceOutput` class"""
+
+    output_format: Optional[OutputFormat] = pd.Field(alias="outputFormat")
+    animation_frequency: Optional[Union[PositiveInt, Literal[-1]]] = pd.Field(
+        alias="animationFrequency"
+    )
+    animation_frequency_offset: Optional[int] = pd.Field(alias="animationFrequencyOffset")
+    output_fields: Optional[List[Union[CommonFieldVars, VolumeSliceFieldVars]]] = pd.Field(
+        alias="outputFields"
+    )
+    slices: Optional[Slices]
+
+    bet_metrics: Optional[bool] = pd.Field(alias="betMetrics")
+    bet_metrics_per_disk: Optional[bool] = pd.Field(alias="betMetricsPerDisk")
+
+
+class SliceOutputPrivate(SliceOutput):
+    """:class:`SliceOutputPrivate` class"""
+
+    coarsen_iterations: Optional[int] = pd.Field(alias="coarsenIterations")
+
+
+class VolumeOutput(OutputBase):
+    """:class:`VolumeOutput` class"""
+
+    output_format: Optional[OutputFormat] = pd.Field(alias="outputFormat")
+    animation_frequency: Optional[Union[PositiveInt, Literal[-1]]] = pd.Field(
+        alias="animationFrequency"
+    )
+    animation_frequency_offset: Optional[int] = pd.Field(alias="animationFrequencyOffset")
+    animation_frequency_time_average: Optional[Union[PositiveInt, Literal[-1]]] = pd.Field(
+        alias="animationFrequencyTimeAverage"
+    )
+    animation_frequency_time_average_offset: Optional[int] = pd.Field(
+        alias="animationFrequencyTimeAverageOffset"
+    )
+    compute_time_averages: Optional[bool] = pd.Field(alias="computeTimeAverages")
+    start_average_integration_step: Optional[int] = pd.Field(alias="startAverageIntegrationStep")
+    output_fields: Optional[List[Union[CommonFieldVars, VolumeSliceFieldVars]]] = pd.Field(
+        alias="outputFields"
+    )
+
+    bet_metrics: Optional[bool] = pd.Field(alias="betMetrics")
+    bet_metrics_per_disk: Optional[bool] = pd.Field(alias="betMetricsPerDisk")
```

Comment:
move all bool options to separate class that will be only used in updater but not be part of allowed configuration

Reply:
Fixed in 05ca5e6, there is quite a bit of complexity in our output data class hierarchy now, but we don't expose it to the user so I don't think its a big issue.